### PR TITLE
boards: efm32: adapt pr feedback for other boards

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description

During the review of #8630 some comments were raised that have slipped through. This PR adapts these comments for the:

* `ikea-tradfri`
* `sltb001a`
* `stk3600`
* `stk3700`

### Issues/PRs references

#7929
#8047  
#8585
#8520

### CODE REVIEW

**Comment by basilfx** [comment]
For what its worth, this is the proof that it still runs after 4ff507a 🤓

SLTB001a:
schermafbeelding 2018-03-16 om 20 20 20

SLSTK3401a:
schermafbeelding 2018-03-16 om 20 23 44

File: boards/sltb001a/board.c [hunk 1]
```diff
@@ -21,9 +21,8 @@
 
 #include "board.h"
 #include "board_common.h"
-#include "pic.h"
-
 #include "periph/gpio.h"
+#include "pic.h"
```
Comment (aabadie) [inline boards/sltb001a/board.c]: Maybe this include could be protected with a `#ifdef MODULE_SILABS_PIC` ?

Comment (basilfx) [reply boards/sltb001a/board.c]: I can add that, but it compiles fine without (if I remove `silabs_pic` from the `USEMODULES`).

Let me know what you prefer.

Comment (basilfx) [reply boards/sltb001a/board.c]: On second thought: I'll add it. That's more similar to how we also include optional peripherals.

**Comment by basilfx** [comment]
Added the conditional includes.

**Pull request merged**